demo-app/src/open-truss/components/TrinoDemo.tsx

```diff
@@ -1,16 +1,20 @@
 'use client'
 import {
+  type BaseOpenTrussComponentV1,
   BaseOpenTrussComponentV1PropsShape,
   withChildren,
 } from '@open-truss/open-truss'
 import { z } from 'zod'
 
-export const Props = withChildren(BaseOpenTrussComponentV1PropsShape).extend({
+export const Props = BaseOpenTrussComponentV1PropsShape.extend({
+  ...withChildren,
   data: z.string().default('no data'),
```

Comment:
This is an interesting prop. `data` is already declared in the `BaseOpenTrussComponentV1Props` type as `YamlType | undefined`. Redeclaring it here effectively extends it to also have `string` as a valid type, hence `(YamlType | undefined) & string` in Keith's screenshot.

I _think_ we may want to go the route of removing data from the base type altogether and not pass in the entire data object to every component. This forces components to add an interface for the data they need and will make composability easier as we have an interface to work off of. Does that seem like a reasonable path to go down? 

Of course we could have a handful of components that legitimately take in the entire data payload, but that would be less common I think. It is a case where the expectation is the component would have a "Data" higher order type as an interface and the entire data payload makes sense for those use cases (e.g. rendering the whole blob to the page for debugging, saving an entire blob off to storage for logging, etc..).

Reply:
Good catch @hktouw -- I thought this was just a dummy prop that @jonmagic added at some point and that it was incorrect (and that it was okay because this component's not done yet). I didn't realize that zod's `.extend` adds a type union. This is a good nudge to consider what you're saying 👍 

> I think we may want to go the route of removing data from the base type altogether and not pass in the entire data object to every component.

After seeing how `.extend` works, I agree. I think we could help with this and add something akin to but more advanced than the `withChildren` I created in this PR that is a function that helps folks create a `data` type that adhere's to what `uqi` will fetch and is typed correctly for the type of query results the component expects.

```ts
export const Props = BaseOpenTrussComponentV1PropsShape.extend({
  data: uqiResultShape({ rowShape: z.object({ accountId: z.number(), login: z.string() }) }),
})

// then, in the component

props.row.map(({ accountId, login }) => {
  // `accountId` has type number
  // `login` has type string
})
```

Pretty sure this is possible to do 😅